ocs_ci/ocs/resources/pod.py

```diff
@@ -1032,11 +1032,14 @@ def get_pvc_name(pod_obj):
         pod_obj (str): The pod object
 
     Returns:
-        pvc_name (str): The pvc_name on a given pod_obj
+        (str): The pvc name of a given pod_obj,
+                None if there is no pvc attached
+
     """
-    return pod_obj.get().get(
-        'spec'
-    ).get('volumes')[0].get('persistentVolumeClaim').get('claimName')
+    pvc = pod_obj.get().get('spec').get('volumes')[0].get('persistentVolumeClaim')
+    if pvc:
+        return pvc.get('claimName')
+    return pvc
```

Comment:
what's the point of this condition? 

It would be better to return the pvc and assert outside, where you call this method

Reply:
tagged you in a comment above

Reply:
thanks. You can do this instead:

```python
pvc = pod_obj.get().get('spec').get('volumes')[0].get('persistentVolumeClaim')
assert pvc, "Failed to find the PVC for pod {pod_obj.get('metadata').get('name')}"
return return pvc.get('claimName')
```

Reply:
and if you do that, the function will not return None, so please remove the line about None from the returned value in the docstring

Reply:
but its not a failure, there will be places where wee want to get None and see that the pod doesent have a pvc attached to it.

Reply:
much easier to check 'None' then trying to catch an assertion error

Reply:
the place that calls this method will not have to check for the assert as the assert is an exception which will be thrown in case the pvc object doesn't exist. 
We can go with either approach, although I don't see the use case in which we'll want to return None

Reply:
when deleting pvcs and pods, we want to check that we don't leave any unused pvc behind

Reply:
or a pvc that will make a pod that needs to be deleted come up again and again